src/rendition-mixin.js

```diff
@@ -87,7 +92,8 @@ let renditionSelectionMixin = function(hlsHandler) {
     return playlists
       .master
       .playlists
-      .map((e, i) => new Representation(hlsHandler, e, i));
+      .filter((media) => !isBlacklisted(media))
```

Comment:
So I added this to filter out media that was blacklisted in `excludeIncompatibleVariants_` so that we don't add quality levels for a playlist that will never be selected. It's a good point that since every time representations is called, it rebuilds its list with this filter, so temporary blacklisting will get removed here. With respect to quality levels this isn't an issue since we only call representations once and then work with the quality level list. Could filter on `media.excludeUntil !== Infinity` instead to remove the permanent blacklisted media but still allow temporary blacklists through.